hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotFileCache.java

```diff
@@ -214,49 +214,55 @@ public void triggerCacheRefreshForTesting() {
 
   private synchronized void refreshCache() throws IOException {
     // get the status of the snapshots directory and check if it is has changes
-    FileStatus dirStatus;
-    try {
-      dirStatus = fs.getFileStatus(snapshotDir);
-    } catch (FileNotFoundException e) {
-      if (this.cache.size() > 0) {
-        LOG.error("Snapshot directory: " + snapshotDir + " doesn't exist");
+    // We need to check the internal folder as Object Stores do not always update
+    // the parent directory's modification time.
+    FileStatus[] snapshotsOnDisk = FSUtils.listStatus(fs, snapshotDir);
+    if (snapshotsOnDisk == null) {
+      // remove all the remembered snapshots because we don't have any left
+      if (LOG.isDebugEnabled() && this.snapshots.size() > 0) {
+        LOG.debug("No snapshots on-disk under: {}, cache empty", snapshotDir);
       }
+      this.snapshots.clear();
+      this.cache.clear();
       return;
     }
 
+    int containsTmpDir = 0;
+    long subDirLastModifiedTime = Long.MIN_VALUE;
```

Comment:
OK, good, this is also what I plan to do for HBASE-22190...

Reply:
:)